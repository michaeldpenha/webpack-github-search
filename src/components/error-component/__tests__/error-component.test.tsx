import React from 'react';
import renderEl from 'src/testUtils/tests-render';
import ErrorComponent from '..';

const errors = {
  message: '',
  name: '',
  networkError: '',
  extraInfo: {},
  graphQLErrors: [
    {
      path: ['repository'],
      locations: [
        {
          line: 2,
          column: 3,
        },
      ],
      message: "Could not resolve to a Repository with the name 'und/react'.",
    },
  ],
};

const setup = ({ errors }) => renderEl(<ErrorComponent errors={errors} />);

describe('Error component', () => {
  it('renders the component wihtout crashing', () => {
    const { getByTestId } = setup({
      errors,
    });

    const el = getByTestId('error-0');
    expect(el).toHaveTextContent("Could not resolve to a Repository with the name 'und/react'.");

    expect(el).toMatchSnapshot();
  });
});
