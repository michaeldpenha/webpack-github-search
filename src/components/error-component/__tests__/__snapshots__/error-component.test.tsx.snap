// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders the component wihtout crashing 1`] = `
<li
  data-testid="error-0"
>
  Could not resolve to a Repository with the name 'und/react'.
</li>
`;
